src/main/java/frc/trigon/robot/motorsimulation/MotorSimulation.java

```diff
@@ -0,0 +1,83 @@
+package frc.trigon.robot.motorsimulation;
+
+import com.ctre.phoenix6.controls.DutyCycleOut;
+import com.ctre.phoenix6.controls.MotionMagicVoltage;
+import com.ctre.phoenix6.controls.PositionVoltage;
+import com.ctre.phoenix6.controls.VoltageOut;
+import edu.wpi.first.math.MathUtil;
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.math.controller.ProfiledPIDController;
+import edu.wpi.first.math.trajectory.TrapezoidProfile;
+import frc.trigon.robot.utilities.Conversions;
+
+public abstract class MotorSimulation {
+    private PIDController pidController;
+    private ProfiledPIDController profiledPIDController;
+    private double voltageCompensationSaturation;
+    private double
+            kS,
+            kG,
+            kV,
+            kA;
+    private double voltage = 0;
+
+    public void applyConfiguration(MotorSimulationConfiguration config) {
+        profiledPIDController = new ProfiledPIDController(config.kP, config.kI, config.kD, new TrapezoidProfile.Constraints(config.maxVelocity, config.maxAcceleration));
+        pidController = new PIDController(config.kP, config.kI, config.kD);
+        voltageCompensationSaturation = config.voltageCompensationSaturation;
+
+        this.kS = config.kS;
+        this.kG = config.kG;
+        this.kV = config.kV;
+        this.kA = config.kA;
+
+        pidController.enableContinuousInput(config.minimumContinuousOutput, config.maximumContinuousOutput);
+    }
+
+    public void stop() {
+        voltage = 0;
+        setInputVoltage(0);
+    }
+
+    public void setVoltage(double voltage) {
+        double compensatedVoltage = MathUtil.clamp(voltage, -voltageCompensationSaturation, voltageCompensationSaturation);
+        voltage = compensatedVoltage;
+        setInputVoltage(compensatedVoltage);
+    }
+
+    public void setControl(VoltageOut VoltageRequest) {
+        setVoltage(VoltageRequest.Output);
+    }
+
+    public void setControl(PositionVoltage positionVoltage) {
+        double voltage = pidController.calculate(positionVoltage.Position);
```

Comment:
What about current position? and you want to compare whatever units are in the conversions factor.

Reply:
again, wdym compare with the conversions factor?